libbeat/cmd/instance/beat.go

```diff
@@ -504,10 +504,6 @@ func (b *Beat) launch(settings Settings, bt beat.Creator) error {
 
 	logp.Info("%s start running.", b.Info.Beat)
 
-	// Launch config manager
-	b.Manager.Start(beater.Stop)
-	defer b.Manager.Stop()
```

Comment:
@blakerouse Look at this code, if I am reading this correctly it mean that when the beats was shutting down we were never changing the state to the **beats** to stopping because no callback was defined see:

```golang
	if cm.stopFunc != nil {
		cm.client.Status(proto.StateObserved_STOPPING, "Stopping", nil)
		cm.stopFunc()
	}
```

So the function was never used as a synchronization mechanism.

Reply:
Correct if not `stopFunc` was defined we would not set it to stopping, which I still think is correct. Because without a `stopFunc` then there is no way for the manager to really signal that.

Reply:
The problem is that signal is only defined before, so I have to set the stopFunc before doing any work downstream in the beat. I get it know, sorry I was confused by the startup sequence and clearly I didn't understand the purpose of it.

Reply:
I think the easiest change would be to either add the StopFunc as part of the contructor of the manager or to add a `SetStopFunc`